src/components/forms/FormFactory.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  FormControl,
+  FormGroup,
+  Col,
+  ControlLabel,
+  HelpBlock,
+  ButtonGroup,
+  DropdownButton,
+  MenuItem,
+  Checkbox,
+  Form
+} from 'patternfly-react';
+import { get, has } from 'lodash';
+import './FormFactory.css';
+
+export const FormFactory = ({ fields, fieldsValues, onFormChange }) => {
+  const formGroups = Object.keys(fields)
+    .filter(key => !(fields[key].isVisible && fields[key].isVisible(fieldsValues) === false))
+    .map(key => {
+      let child;
+      const validMsg = get(fieldsValues[key], 'validMsg', null);
+      switch (fields[key].type) {
+        case 'textarea':
+          child = (
+            <FormControl
```

Comment:
does it make sense to extract each case statement to its own component? i could see reusing them in possibility in other context + i believe it would be easier to read?